src/login/logind-session-device.c

```diff
@@ -417,9 +422,10 @@ void session_device_free(SessionDevice *sd) {
                         (void) sd_notify(false, m);
         }
 
-        session_device_stop(sd);
+        close_fd = session_device_stop(sd);
         session_device_notify(sd, SESSION_DEVICE_RELEASE);
-        safe_close(sd->fd);
+        if (close_fd)
+                safe_close(sd->fd);
```

Comment:
not sure i follow... safe_close() is a no-op anyway if sd->fd is negative, so what does this precisely change?